test/expect-tests/dune_patch/dune_patch_tests.ml

```diff
@@ -199,10 +199,21 @@ let%expect_test "patching a new file" =
 let () = Dune_util.Report_error.report_backtraces true
 
 let%expect_test "patching a deleted file" =
-  test [ "foo.ml", "This is wrong\n" ] ("foo.patch", delete_file);
-  check "foo.ml";
-  [%expect {|
-    File foo.ml not found |}]
+  let filename = "foo.ml" in
+  test [ filename, "This is wrong\n" ] ("foo.patch", delete_file);
+  (* Different implementations of the patch command behave differently when the
+     patch specifies deleting a file: *)
+  match (Unix.stat filename).st_kind with
+  | S_REG ->
+    (* Some implementations of patch (e.g. the patch that ships with macOS
+       15.1) do not delete files, and instread truncate them to 0 length. If
+       the file still exists after applying the patch, assert that it is now
+       empty. *)
+    assert (String.length (Io.String_path.read_file filename) == 0)
```

Comment:
I think instead of reading the file I think it would be better to look at the `stat` metadata and read `st_size`. We already `stat` above and reading the size feels more direct than checking the length of the contents we read.

Reply:
Good idea. Done